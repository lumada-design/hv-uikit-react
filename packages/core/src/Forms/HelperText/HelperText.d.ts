import { StandardProps } from "@material-ui/core";

export interface HvInfoTextProps
  extends StandardProps<React.HTMLAttributes<HTMLDivElement>, HvInfoTextClassKey> {
  /**
   * The text to be shown by the helper text as a notification for the user.
   */
  notification?: string;
  /**
   * If ´true´ the input is disabled.
   */
  disabled?: boolean;
}

export type HvInfoTextClassKey = "helperText" | "showText" | "helperDisabled" | "@global";

export default function HvInfoText(props: HvInfoTextProps): JSX.Element | null;
