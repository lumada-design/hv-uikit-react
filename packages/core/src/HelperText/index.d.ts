export { default } from "./HelperText";
export * from "./HelperText";
