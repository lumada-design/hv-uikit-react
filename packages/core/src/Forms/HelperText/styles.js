const styles = theme => ({
  root: {},
  topGutter: { marginTop: 8 },
  helperDisabled: {
    color: theme.hv.palette.atmosphere.atmo7,
    display: "none"
  },
  helperText: {
    color: theme.hv.palette.accent.acce1,
    display: "none"
  },
  showText: {
    display: "block"
  },
  "@global": {
    "input:-webkit-autofill": {
      "-webkit-box-shadow": `0 0 0px 1000px ${theme.hv.palette.atmosphere.atmo1} inset`,
      "-webkit-text-fill-color": theme.hv.typography.normalText.color
    }
  }
});

export default styles;
