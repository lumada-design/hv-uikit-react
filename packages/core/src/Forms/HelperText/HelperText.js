import React from "react";
import PropTypes from "prop-types";
import clsx from "clsx";
import { withStyles } from "@material-ui/core";
import { isEmpty } from "lodash";
import { HvFormElementContextConsumer } from "../FormElement";
import HvTypography from "../../Typography";

import styles from "./styles";
/**
 * Component used in conjunction with other form elements, to give extra information about status.
 */
const HvHelperText = props => {
  const {
    children,
    notification = "",
    classes,
    id,
    disabled,
    disableGutter = false,
    ...others
  } = props;

  return (
    <HvFormElementContextConsumer>
      {formContext => {
        const { elementDisabled } = formContext;
        const isVisible = !isEmpty(notification);
        const localDisabled = disabled || elementDisabled;
        return (
          <>
            <HvTypography
              id={id}
              variant="infoText"
              className={clsx({
                [classes.showText]: !isVisible,
                [classes.helperDisabled]: localDisabled,
                [classes.helperText]: !localDisabled,
                [classes.topGutter]: !disableGutter
              })}
              {...others}
            >
              {children}
            </HvTypography>
            <HvTypography
              id={`${id}-notification`}
              variant="infoText"
              className={clsx({
                [classes.showText]: isVisible,
                [classes.helperDisabled]: localDisabled,
                [classes.helperText]: !localDisabled,
                [classes.topGutter]: !disableGutter
              })}
              aria-live="polite"
              aria-atomic="true"
              aria-relevant="additions text"
              {...others}
            >
              {notification}
            </HvTypography>
          </>
        );
      }}
    </HvFormElementContextConsumer>
  );
};

HvHelperText.propTypes = {
  /**
   * The description to be shown by this helper text
   */
  children: PropTypes.node.isRequired,
  /**
   * Id to be applied to the root node.
   */
  id: PropTypes.string,
  /**
   * A Jss Object used to override or extend the component styles applied.
   */
  classes: PropTypes.shape({
    /**
     * Styles applied to the information text.
     */
    helperText: PropTypes.string,
    /**
     * Styles applied when the text should be shown.
     */
    showText: PropTypes.string,
    /**
     * Styles applied when the text is disabled.
     */
    helperDisabled: PropTypes.string,
    /**
     * Separation between text and upper element.
     */
    topGutter: PropTypes.string,
    /**
     * IE11 specific styling.
     */
    "@global": PropTypes.string
  }).isRequired,
  /**
   * The text to replace the description to shown an alert to the user.
   */
  notification: PropTypes.string,
  /**
   * If ´true´ the text is disabled.
   */
  disabled: PropTypes.bool,
  /**
   * If ´true´ the text won't include a top gutter.
   */
  disableGutter: PropTypes.bool
};

export default withStyles(styles, { name: "HvHelperText" })(HvHelperText);
