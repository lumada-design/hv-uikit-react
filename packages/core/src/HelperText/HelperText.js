import React from "react";
import PropTypes from "prop-types";
import isNil from "lodash/isNil";
import clsx from "clsx";
import { Input, withStyles } from "@material-ui/core";
import InfoS from "@hv/uikit-react-icons/dist/Info";
import withId from "../withId";
import withLabels from "../withLabels";
import { isKeypress, KeyboardCodes } from "../utils";
import isBrowser from "../utils/browser";
import InputAdornment from "./InputAdornment";
import HvTypography from "../Typography";
import HvList from "../List";
import validationTypes from "./validationTypes";

import { validateCharLength, validateInput } from "./validations";
import styles from "./styles";
import withTooltips from "../withTooltip";

const DEFAULT_LABELS = {
  inputLabel: "",
  placeholder: "",
  infoText: "",
  warningText: "something wrong",
  maxCharQuantityWarningText: "The value is too big",
  minCharQuantityWarningText: "The value is too short",
  requiredWarningText: "The value is required",
  clearButtonLabel: "Clear the text"
};

const validationState = {
  empty: "empty",
  filled: "filled",
  valid: "valid",
  invalid: "invalid"
};

/**
 * An input is a graphicl control element that allows the user to write text.
 */
const HvHelperText = props => {
  const {
    labels,
    classes,
    className,
    id,
    password,
    disabled,
    isRequired,
    infoIcon,
    validationIconVisible,
    disableClear,
    customFixedIcon,
    validationIconPosition,
    showInfo,
    validationType,
    validationState,
    maxCharQuantity,
    minCharQuantity,
    validation,
    externalWarningTextOverride,
    value,
    autoFocus,
    initialValue,
    inlineStyling,
    variant,
    classesToApply,
    hasIcon,
    stateValidation,
    ...others
  } = props;

  const getTypographyType = chosenVariant => {
    return chosenVariant === "helper"
      ? "infoText"
      : chosenVariant === "warning"
      ? "sText"
      : "sText";
  };

  const getClassesToApply = chosenVariant => {
    let labelStyles;
    if (chosenVariant === "helper") {
      labelStyles = clsx(classes.infoText);
    } else if (chosenVariant === "warning") {
      labelStyles = clsx(classes.textWarning, classes.infoText, {
        // show the text if we are in an invalid state and either of the invalid labels exist
        [classes.showText]:
          stateValidation === "invalid" && (externalWarningTextOverride || labels.warningText)
      });
    }
    return labelStyles;
  };

  const getLabels = chosenVariant => {
    let labelToReturn;
    if (chosenVariant === "helper") {
      labelToReturn = labels.infoText;
    } else if (chosenVariant === "warning") {
      labelToReturn = externalWarningTextOverride || labels.warningText || "";
    }
    return labelToReturn;
  };

  // const getInlineStyles = (errorIconExists, stateValidationObject) => {
  //   return !errorIconExists && stateValidationObject !== validationState.invalid
  //     ? "block"
  //     : "none";
  // };

  let labelType;
  // const currentInputState = hasIcon && stateValidation !== validationState.invalid;
  if (stateValidation !== "invalid") {
    labelType = variant;
  } else {
    labelType = "warning";
  }
  return (
    <>
      <HvTypography
        id={`${id}-description`}
        variant={getTypographyType(labelType)}
        className={getClassesToApply(labelType, stateValidation)}
        // style={{
        //   display:
        //     !infoIcon && stateValidationState !== validationStates.invalid ? "block" : "none"
        // }}
      >
        {/* Blah */}
        {getLabels(labelType)}
      </HvTypography>
      {/* {showInfo && labels.infoText && ( */}
      {/* <HvTypography
          id={`${id}-description`}
          variant={getTypographyType(variant)}
          className={getClassesToApply(variant)}
          // style={{
          //   display: getInlineStyles(hasIcon, stateValidation)
          // }}
        >
          {getLabels(variant)}
        </HvTypography> */}
      {/* )} */}
      {/* <HvTypography
          variant="sText"
          className={clsx(classes.textWarning, classes.infoText, {
            [classes.showText]:
              stateValidationState === validationStates.invalid &&
              (externalWarningTextOverride || warningText)
          })}
          aria-live="polite"
          aria-controls={`${id}-input`}
          aria-atomic="true"
          aria-relevant="additions text"
          aria-labelledby={labels.inputLabel ? `${id}-label` : null}
        >
          {externalWarningTextOverride || warningText || ""}
        </HvTypography> */}
    </>
  );
};

HvHelperText.propTypes = {
  inlineStyling: PropTypes.shape({}),
  variant: PropTypes.string,
  classesToApply: PropTypes.string,
  hasIcon: PropTypes.bool,
  stateValidation: PropTypes.shape({}),
  /**
   * Class names to be applied.
   */
  className: PropTypes.string,
  /**
   * Id to be applied to the root node.
   */
  id: PropTypes.string,
  /**
   * A Jss Object used to override or extend the component styles applied.
   */
  classes: PropTypes.shape({
    /**
     * Styles applied to the root container of the input.
     */
    root: PropTypes.string,
    /**
     * Styles applied to input root which is comprising of everything but the labels and descriptions.
     */
    inputRoot: PropTypes.string,
    /**
     * Styles applied to input root when it is disabled.
     */
    inputRootDisabled: PropTypes.string,
    /**
     * Styles applied to input root when it is invalid.
     */
    inputRootInvalid: PropTypes.string,
    /**
     * Styles applied to input root when it is focused.
     */
    inputRootFocused: PropTypes.string,
    /**
     * Styles applied to input html element.
     */
    input: PropTypes.string,
    /**
     * Styles applied to input html element when it is disabled.
     */
    inputDisabled: PropTypes.string,
    /**
     * Styles applied to the container of the suggestions list.
     */
    suggestionsContainer: PropTypes.string,
    /**
     * Styles applied to the suggestions list.
     */
    suggestionList: PropTypes.string,
    /**
     * Styles applied to input html element when it is multiline mode.
     */
    multiLine: PropTypes.string,
    /**
     * Styles applied to the label element.
     */
    label: PropTypes.string,
    /**
     * Styles applied to the label element when it is disabled.
     */
    labelDisabled: PropTypes.string,
    /**
     * Styles applied to the container of the labels elements.
     */
    labelContainer: PropTypes.string,
    /**
     * Styles applied to the icon information container.
     */
    infoIconContainer: PropTypes.string,
    /**
     * Styles applied to the icon information text.
     */
    infoText: PropTypes.string,
    /**
     * Styles applied to the description.
     */
    text: PropTypes.string,
    /**
     * Styles applied to the description when it is showing an information.
     */
    textInfo: PropTypes.string,
    /**
     * Styles applied to the description when it is showing a warning.
     */
    textWarning: PropTypes.string,
    /**
     * Styles applied when the text should be shown.
     */
    showText: PropTypes.string,
    /**
     * Styles applied to the input adornment icons.
     */
    icon: PropTypes.string,
    /**
     * Styles applied to the icon used to clean the input.
     */
    iconClear: PropTypes.string,
    /**
     * IE11 specific styling.
     */
    "@global": PropTypes.string
  }).isRequired,
  /**
   * An Object containing the various texts associated with the input.
   */
  labels: PropTypes.shape({
    /**
     * The label on top of the input.
     */
    inputLabel: PropTypes.string,
    /**
     * The placeholder value of the input.
     */
    placeholder: PropTypes.string,
    /**
     * The default value of the info text below the input.
     */
    infoText: PropTypes.string,
    /**
     * The value when a validation fails.
     */
    warningText: PropTypes.string,
    /**
     * The message that appears when there are too many characters.
     */
    maxCharQuantityWarningText: PropTypes.string,
    /**
     * The message that appears when there are too few characters.
     */
    minCharQuantityWarningText: PropTypes.string,
    /**
     * The message that appears when the input is empty and required.
     */
    requiredWarningText: PropTypes.string,
    /**
     * The label of the clear button.
     */
    clearButtonLabel: PropTypes.string
  }),
  /**
   * Attributes applied to the input element.
   */
  inputProps: PropTypes.instanceOf(Object),
  /**
   * Allows passing a ref to the underlying input
   */
  inputRef: PropTypes.shape({ current: PropTypes.any }),
  /**
   * If ´true´ the input is disabled.
   */
  disabled: PropTypes.bool,
  /**
   * If ´true´ the input value must be filled on blur or else the validation fails.
   */
  isRequired: PropTypes.bool,
  /**
   * If ´true´ the input is of type password hiding the value.
   */
  password: PropTypes.bool,
  /**
   * The function that will be executed onChange, allows modification of the input,
   * it receives the value. If a new value should be presented it must returned it.
   */
  onChange: PropTypes.func,
  /**
   * The function that will be executed onBlur, allows checking the validation state,
   * it receives the value and the validation state (´empty´, ´filled´, ´invalid´, ´valid´).
   */
  onBlur: PropTypes.func,
  /**
   * The function that will be executed onBlur, allows checking the value state,
   * it receives the value.
   */
  onFocus: PropTypes.func,
  /**
   * The function that will be executed onKeyDown, allows checking the value state,
   * it receives the event and value.
   */
  onKeyDown: PropTypes.func,
  /**
   * The function that will be executed to received an array of objects that has a label and id to create list of suggestion
   */
  suggestionListCallback: PropTypes.func,
  /**
   * The function that will be executed after selecting a value in the suggestion list
   */
  suggestionSelectedCallback: PropTypes.func,
  /**
   * If `true` information label is shown, `false` otherwise.
   */
  showInfo: PropTypes.bool,
  /**
   * The custom validation function, it receives the value and must return
   * either ´true´ for valid or ´false´ for invalid, default validations would only
   * occur if this function is null or undefined
   */
  validation: PropTypes.func,
  /**
   * The value of the input, when controlled.
   */
  value: PropTypes.string,
  /**
   * The initial value of the input, when uncontrolled.
   */
  initialValue: PropTypes.string,
  /**
   * If `true` it should autofocus.
   */
  autoFocus: PropTypes.bool,
  /**
   * The initial state of the input.
   * note: Is recommended you use the provided validationStates object to set this value.
   */
  validationState: PropTypes.oneOf(["empty", "filled", "invalid", "valid"]),
  /**
   * Show info icon with info label.infoText.
   */
  infoIcon: PropTypes.bool,
  /**
   * If `true` the validation icon is visible, `false` otherwise
   */
  validationIconVisible: PropTypes.bool,
  /**
   * If `true` the clear button is disabled if `false` is enable
   */
  disableClear: PropTypes.bool,
  /**
   * The icon position of the input. It is recommended to use the provided validationIconPosition object to set this value.
   */
  validationIconPosition: PropTypes.oneOf(["left", "right"]),
  /**
   * a custom icon to be added into the input.
   */
  customFixedIcon: PropTypes.node,
  /**
   * The maximum allowed length of the characters, if this value is null no check
   * will be performed.
   */
  maxCharQuantity: PropTypes.number,
  /**
   * The minimum allowed length of the characters, if this value is null no check
   * will be perform.
   */
  minCharQuantity: PropTypes.number,
  /**
   * Which type of default validation should the input perform. It is recommended to use the provided ValidationTypes object to set this value.
   */
  validationType: PropTypes.oneOf(["none", "number", "email"]),
  /**
   * Overrides any validation with a specific error/warning message to set in the warningText slot.
   */
  externalWarningTextOverride: PropTypes.string
};

HvHelperText.defaultProps = {
  className: "",
  id: undefined,
  password: false,
  inputProps: {},
  inputRef: null,
  customFixedIcon: null,
  infoIcon: false,
  validationIconVisible: true,
  disableClear: false,
  validationIconPosition: "right",
  showInfo: true,
  validation: null,
  maxCharQuantity: null,
  minCharQuantity: null,
  validationType: "none",
  value: undefined,
  initialValue: undefined,
  autoFocus: false,
  validationState: validationState.empty,
  disabled: false,
  isRequired: false,
  suggestionListCallback: () => {},
  suggestionSelectedCallback: () => {},
  onChange: (event, value) => value,
  onBlur: () => {},
  onFocus: () => {},
  onKeyDown: () => {},
  externalWarningTextOverride: null
};

export default withStyles(styles, { name: "HvHelperText" })(
  withLabels(DEFAULT_LABELS)(withId(HvHelperText))
);
