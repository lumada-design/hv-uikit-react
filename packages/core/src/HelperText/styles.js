import { outlineStyles } from "../Focus/styles";

const styles = theme => ({
  root: {
    minWidth: "150px",
    maxWidth: "610px"
  },
  inputRoot: {
    margin: "0",
    width: "100%",
    borderStyle: "solid",
    borderWidth: " 0 0 1px 0",
    background: theme.hv.palette.atmosphere.atmo1,
    borderColor: theme.hv.palette.atmosphere.atmo6,
    "&:hover": {
      borderStyle: "solid",
      borderWidth: " 0 0 1px 0",
      borderColor: theme.hv.palette.accent.acce1,
      background: theme.hv.palette.atmosphere.atmo1
    },
    "&:hover, &:focus-within": {
      "& $iconClear": {
        visibility: "visible"
      }
    }
  },
  inputRootDisabled: {
    borderColor: theme.hv.palette.atmosphere.atmo6,
    background: theme.hv.palette.atmosphere.atmo4,
    "&:hover": {
      borderColor: theme.hv.palette.atmosphere.atmo6,
      background: theme.hv.palette.atmosphere.atmo4,
      cursor: "not-allowed"
    },
    cursor: "not-allowed"
  },
  inputRootFocused: {
    borderStyle: "solid",
    borderWidth: " 0 0 1px 0",
    borderColor: theme.hv.palette.accent.acce1,
    background: theme.hv.palette.atmosphere.atmo1,
    ...outlineStyles,
    "&:hover": {
      borderStyle: "solid",
      borderWidth: "0 0 1px 0",
      borderColor: theme.hv.palette.accent.acce1,
      background: theme.hv.palette.atmosphere.atmo1
    },
    "& $iconClear": { visibility: "visible" }
  },
  inputRootInvalid: {
    borderColor: theme.hv.palette.semantic.sema4,
    "&:hover": {
      borderColor: theme.hv.palette.semantic.sema4
    }
  },
  input: {
    height: `20px`,
    marginLeft: `${theme.hv.spacing.xs}px`,
    marginRight: `${theme.hv.spacing.xs}px`,
    padding: "6px 0 5px",
    ...theme.hv.typography.normalText,
    overflow: "hidden",
    textOverflow: "ellipsis",
    "&::placeholder": {
      ...theme.hv.typography.placeholderText,
      opacity: 1
    },
    "&::-ms-clear": {
      display: "none"
    }
  },
  inputDisabled: {
    cursor: "not-allowed"
  },
  multiLine: {
    padding: 0
  },
  labelContainer: {
    display: "flex",
    alignItems: "flex-end",
    justifyContent: "space-between"
  },
  label: {
    paddingBottom: "8px",
    display: "block"
  },
  labelDisabled: {
    color: theme.hv.palette.atmosphere.atmo7
  },
  infoIconContainer: {
    height: "32px",
    width: "32px",
    display: "flex",
    justifyContent: "center",
    "& div": {
      alignSelf: "center",
      marginTop: -2
    }
  },
  infoText: {
    color: theme.hv.palette.accent.acce1,
    paddingTop: "8px",
    display: "block"
  },
  textWarning: {
    color: theme.hv.palette.accent.acce1,
    display: "none"
  },
  showText: {
    display: "block"
  },
  adornmentsBox: {
    display: "flex",
    flexDirection: "row",
    height: 30,
    justifyContent: "center"
  },
  adornmentButton: {
    backgroundColor: "transparent",
    border: "none",
    padding: 0,
    margin: 0,
    cursor: "pointer"
  },
  icon: {
    width: 30,
    height: 30
  },
  iconClear: {
    visibility: "hidden"
  },

  suggestionsContainer: {
    width: "100%",
    position: "relative",
    top: "-1px"
  },
  suggestionList: {
    backgroundColor: theme.hv.palette.atmosphere.atmo1,
    boxShadow: "0 0 0 #fff, 0px -5px 12px -5px rgba(65,65,65,.12)",
    padding: `${theme.hv.spacing.sm}px`,
    position: "absolute",
    width: "100%"
  },
  "@global": {
    "input:-webkit-autofill": {
      "-webkit-box-shadow": `0 0 0px 1000px ${theme.hv.palette.atmosphere.atmo1} inset`,
      "-webkit-text-fill-color": theme.hv.typography.normalText.color
    }
  }
});

export default styles;
