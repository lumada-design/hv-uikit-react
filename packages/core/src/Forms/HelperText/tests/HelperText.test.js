/* eslint-env jest */

import React from "react";
import { mount } from "enzyme";

import HvProvider from "../../../Provider";
import HelperText from "../HelperText";

/* eslint-disable no-console */

describe("HelperText", () => {
  let wrapper;

  beforeEach(async () => {
    wrapper = mount(
      <HvProvider>
        <HelperText id="base" notification="test">
          test
        </HelperText>
      </HvProvider>
    );
  });

  it("should be defined", () => {
    expect(wrapper).toBeDefined();
  });

  it("should render correctly", () => {
    expect(wrapper.find(HelperText)).toMatchSnapshot();
  });

  it("should render the Input component", () => {
    const HvHelperTextComponent = wrapper.find(HelperText);
    expect(HvHelperTextComponent.length).toBe(1);
  });
});
