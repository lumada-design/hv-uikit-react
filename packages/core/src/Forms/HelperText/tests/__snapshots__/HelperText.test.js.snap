// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelperText should render correctly 1`] = `
<WithStyles(HvHelperText)
  id="base"
  notification="test"
>
  <HvHelperText
    classes={
      Object {
        "helperDisabled": "HvHelperText-helperDisabled",
        "helperText": "HvHelperText-helperText",
        "root": "HvHelperText-root",
        "showText": "HvHelperText-showText",
        "topGutter": "HvHelperText-topGutter",
      }
    }
    id="base"
    notification="test"
  >
    <WithStyles(ForwardRef)
      className="HvHelperText-helperText HvHelperText-topGutter"
      id="base"
      variant="infoText"
    >
      <ForwardRef
        className="HvHelperText-helperText HvHelperText-topGutter"
        classes={
          Object {
            "3xlTitle": "HvTypography-3xlTitle",
            "4xlTitle": "HvTypography-4xlTitle",
            "5xlTitle": "HvTypography-5xlTitle",
            "disabledButtonText": "HvTypography-disabledButtonText",
            "highlightText": "HvTypography-highlightText",
            "infoText": "HvTypography-infoText",
            "inlineLink": "HvTypography-inlineLink",
            "lTitle": "HvTypography-lTitle",
            "labelText": "HvTypography-labelText",
            "mTitle": "HvTypography-mTitle",
            "normalText": "HvTypography-normalText",
            "placeholderText": "HvTypography-placeholderText",
            "root": "HvTypography-root",
            "sLink": "HvTypography-sLink",
            "sText": "HvTypography-sText",
            "sTitle": "HvTypography-sTitle",
            "selectedNavText": "HvTypography-selectedNavText",
            "selectedText": "HvTypography-selectedText",
            "vizText": "HvTypography-vizText",
            "xlTitle": "HvTypography-xlTitle",
            "xsTitle": "HvTypography-xsTitle",
            "xxlTitle": "HvTypography-xxlTitle",
            "xxsTitle": "HvTypography-xxsTitle",
          }
        }
        id="base"
        variant="infoText"
      >
        <p
          className="HvHelperText-helperText HvHelperText-topGutter HvTypography-root HvTypography-infoText"
          id="base"
        >
          test
        </p>
      </ForwardRef>
    </WithStyles(ForwardRef)>
    <WithStyles(ForwardRef)
      aria-atomic="true"
      aria-live="polite"
      aria-relevant="additions text"
      className="HvHelperText-helperText HvHelperText-showText HvHelperText-topGutter"
      id="base-notification"
      variant="infoText"
    >
      <ForwardRef
        aria-atomic="true"
        aria-live="polite"
        aria-relevant="additions text"
        className="HvHelperText-helperText HvHelperText-showText HvHelperText-topGutter"
        classes={
          Object {
            "3xlTitle": "HvTypography-3xlTitle",
            "4xlTitle": "HvTypography-4xlTitle",
            "5xlTitle": "HvTypography-5xlTitle",
            "disabledButtonText": "HvTypography-disabledButtonText",
            "highlightText": "HvTypography-highlightText",
            "infoText": "HvTypography-infoText",
            "inlineLink": "HvTypography-inlineLink",
            "lTitle": "HvTypography-lTitle",
            "labelText": "HvTypography-labelText",
            "mTitle": "HvTypography-mTitle",
            "normalText": "HvTypography-normalText",
            "placeholderText": "HvTypography-placeholderText",
            "root": "HvTypography-root",
            "sLink": "HvTypography-sLink",
            "sText": "HvTypography-sText",
            "sTitle": "HvTypography-sTitle",
            "selectedNavText": "HvTypography-selectedNavText",
            "selectedText": "HvTypography-selectedText",
            "vizText": "HvTypography-vizText",
            "xlTitle": "HvTypography-xlTitle",
            "xsTitle": "HvTypography-xsTitle",
            "xxlTitle": "HvTypography-xxlTitle",
            "xxsTitle": "HvTypography-xxsTitle",
          }
        }
        id="base-notification"
        variant="infoText"
      >
        <p
          aria-atomic="true"
          aria-live="polite"
          aria-relevant="additions text"
          className="HvHelperText-helperText HvHelperText-showText HvHelperText-topGutter HvTypography-root HvTypography-infoText"
          id="base-notification"
        >
          test
        </p>
      </ForwardRef>
    </WithStyles(ForwardRef)>
  </HvHelperText>
</WithStyles(HvHelperText)>
`;
